pkg/executables/cmk.go

```diff
@@ -0,0 +1,308 @@
+	}
+	accounts := response.CmkAccounts
+	if len(accounts) > 1 {
+		return fmt.Errorf("duplicate account %s found", account)
+	} else if len(accounts) == 0 {
+		return fmt.Errorf("account %s not found", account)
+	}
+	return nil
+}
+
+func NewCmk(executable Executable, writer filewriter.FileWriter) *Cmk {
+	return &Cmk{
+		writer:     writer,
+		executable: executable,
+		retrier:    retrier.NewWithMaxRetries(maxRetriesCmk, backOffPeriodCmk),
+	}
+}
+
+// ValidateCloudStackConnection Calls `cmk sync` to ensure that the endpoint and credentials + domain are valid
+func (c *Cmk) ValidateCloudStackConnection(ctx context.Context, config CmkExecConfig) error {
+	buffer, err := c.exec(ctx, config, "sync")
+	if err != nil {
+		return fmt.Errorf("error validating cloudstack connection for cmk config %s: %v", buffer.String(), err)
+	}
+	logger.MarkPass("Connected to CloudStack server")
+	return nil
+}
+
+func (c *Cmk) execWithNameAndIdFilters(ctx context.Context, config CmkExecConfig, parameterValue string, genericArgs ...string) (stdout bytes.Buffer, err error) {
+	argsWithNameFilterArg := append(genericArgs, fmt.Sprintf("name=\"%s\"", parameterValue))
+	result, err := c.exec(ctx, config, argsWithNameFilterArg...)
+	if err != nil {
+		return result, fmt.Errorf("error getting resource info filtering by id %s: %v", parameterValue, err)
+	}
+	if result.Len() == 0 {
+		argsWithIdFilterArg := append(genericArgs, fmt.Sprintf("id=\"%s\"", parameterValue))
+		logger.V(6).Info("No resources found searching by name. Trying again filtering by id instead", "searchParameterValue", parameterValue)
+		result, err = c.exec(ctx, config, argsWithIdFilterArg...)
+		if err != nil {
+			return result, fmt.Errorf("error getting resource info filtering by id %s: %v", parameterValue, err)
+		}
+	}
+	return result, nil
+}
+
+func (c *Cmk) exec(ctx context.Context, config CmkExecConfig, args ...string) (stdout bytes.Buffer, err error) {
+	if err != nil {
+		return bytes.Buffer{}, fmt.Errorf("failed get environment map: %v", err)
+	}
+	configFile, err := c.buildCmkConfigFile(config)
+	if err != nil {
+		return bytes.Buffer{}, fmt.Errorf("failed cmk validations: %v", err)
+	}
+	argsWithConfigFile := append([]string{"-c", configFile}, args...)
+
+	return c.executable.Execute(ctx, argsWithConfigFile...)
+}
+
+type CmkExecConfig struct {
+	ApiKey        string // Api Key for CloudMonkey to access CloudStack Cluster
+	SecretKey     string // Secret Key for CloudMonkey to access CloudStack Cluster
+	ManagementUrl string // Management Endpoint Url for CloudMonkey to access CloudStack Cluster
+	VerifyCert    bool   // boolean indicating if CloudMonkey should verify the cert presented by the CloudStack Management Server
+}
+
+// TODO: Add support for passing in domain from Deployment Config Spec
+func (c *Cmk) buildCmkConfigFile(config CmkExecConfig) (configFile string, err error) {
+	t := templater.New(c.writer)
+	data := map[string]string{
```

Comment:
If you name the "variables" in the template with the same name as the struct fields, you can pass it directly without having to create a map

Reply:
Good point, refactored the struct